csharp/ql/src/experimental/Security Features/JsonWebTokenHandler/JsonWebTokenHandlerLib.qll

```diff
@@ -0,0 +1,265 @@
+  exists(FlowsToTokenValidationResultIsValidCall config, DataFlow::Node source |
+    call = source.asExpr()
+  |
+    config.hasFlow(source, _)
+  )
+}
+
+/**
+ * A property write for security-sensitive properties for `Microsoft.IdentityModel.Tokens.TokenValidationParameters`
+ */
+class TokenValidationParametersPropertyWriteToValidationDelegated extends PropertyWrite {
+  TokenValidationParametersPropertyWriteToValidationDelegated() {
+    exists(Property p, Class c |
+      c.hasQualifiedName("Microsoft.IdentityModel.Tokens.TokenValidationParameters")
+    |
+      p.getAnAccess() = this and
+      c.getAProperty() = p and
+      p.getName() in [
+          "SignatureValidator", "TokenReplayValidator", "AlgorithmValidator", "AudienceValidator",
+          "IssuerSigningKeyValidator", "LifetimeValidator"
+        ]
+    )
+  }
+}
+
+/**
+ * Holds if the callable has a return statement and it always returns true for all such statements
+ */
+predicate callableHasAReturnStmtAndAlwaysReturnsTrue(Callable c) {
+  c.getReturnType().toString() = "Boolean" and
+  forall(ReturnStmt rs | rs.getEnclosingCallable() = c |
+    rs.getChildExpr(0).(BoolLiteral).getBoolValue() = true
+  ) and
+  exists(ReturnStmt rs | rs.getEnclosingCallable() = c)
+}
+
+/**
+ * Holds if the lambda expression `le` always returns true
+ */
+predicate lambdaExprReturnsOnlyLiteralTrue(LambdaExpr le) {
+  le.getExpressionBody().(BoolLiteral).getBoolValue() = true
+}
+
+class CallableAlwaysReturnsTrue extends Callable {
+  CallableAlwaysReturnsTrue() {
+    callableHasAReturnStmtAndAlwaysReturnsTrue(this)
+    or
+    lambdaExprReturnsOnlyLiteralTrue(this)
+    or
+    exists(LambdaExpr le, Call call, CallableAlwaysReturnsTrue cat | this = le |
+      call = le.getExpressionBody() and
+      cat.getACall() = call
+    )
+  }
+}
+
+/**
+ * Holds if any exception being thrown by the callable is of type `System.ArgumentNullException`
+ * It will also hold if no exceptions are thrown by the callable
+ */
+predicate callableOnlyThrowsArgumentNullException(Callable c) {
+  forall(ThrowElement thre | c = thre.getEnclosingCallable() |
+    thre.getThrownExceptionType().hasQualifiedName("System.ArgumentNullException")
+  )
+}
+
+/**
+ * A specialization of `CallableAlwaysReturnsTrue` that takes into consideration exceptions being thrown for higher precision.
+ */
+class CallableAlwaysReturnsTrueHigherPrecision extends CallableAlwaysReturnsTrue {
+  CallableAlwaysReturnsTrueHigherPrecision() {
+    callableOnlyThrowsArgumentNullException(this) and
+    (
+      forall(Call call, Callable callable | call.getEnclosingCallable() = this |
+        callable.getACall() = call and
+        callable instanceof CallableAlwaysReturnsTrueHigherPrecision
+      )
+      or
+      exists(LambdaExpr le, Call call, CallableAlwaysReturnsTrueHigherPrecision cat | this = le |
+        call = le.getExpressionBody() and
```

Comment:
Changing this line to `le.canReturn(call)` would cover a wider scope.

Reply:
fixing it